### PR TITLE
General dependency upgrades (awx venv only for now)

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
This implements a process change and does a general upgrade of everything in the AWX virtual environment which won't break things.

Like all dependency upgrades, most of the real work involves categorizing which things will break things on upgrade. To do this in a more maintainable way, I've written blurbs on those things we can't upgrade, and then _unpinned_ the content in `requirements.in` for those that don't break things.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
9.0.1
```

##### ADDITIONAL INFORMATION

This needs some corresponding changes to unblock upgrade of pip & setuptools, but I have been testing it with temporary hacks for that and everything is going fine, so it looks like the ecosystem has worked out enough to not error here.

### CODE REVIEW

**Comment by AlanCoding** [comment]
New lib license links:

https://pypi.org/project/zipp/
https://github.com/jaraco/zipp/blob/master/LICENSE

https://github.com/tobgu/pyrsistent/blob/master/LICENCE.mit
https://pypi.org/project/pyrsistent/

https://bitbucket.org/ruamel/yaml.clib/src/default/LICENSE
https://pypi.org/project/ruamel.yaml.clib/

https://github.com/tobgu/pyrsistent/blob/master/LICENCE.mit
https://pypi.org/project/pyrsistent/

https://gitlab.com/python-devs/importlib_resources/blob/master/LICENSE
https://pypi.org/project/importlib_resources/

**Comment by AlanCoding** [comment]
Looks like the swagger failure was introduced by an upgrade of DRF in this between the two versions

https://github.com/encode/django-rest-framework/compare/3.9.4...3.10.0

and I know that it was introduced _before_ 7762aaa90f92ff30e521931db2ec65c52f8da98e, still searching locally.

**Comment by AlanCoding** [comment]
passes with

https://github.com/encode/django-rest-framework/commit/9811a29a5a1348f1a5de30a9a7a5d0f8d2fd4843

fails with

https://github.com/encode/django-rest-framework/commit/c3a13916d2c8025b042a9b9fb1ec5af25f202cb4

I think I'm confusing myself because they did branch merges, and I'm only interested in the 3.10 branch

**Comment by AlanCoding** [comment]
introduced in https://github.com/encode/django-rest-framework/commit/37f210a455cc92cb3f61a23e194a1d0de58d149b

> Added OpenAPI Schema Generation.

https://github.com/encode/django-rest-framework/pull/6532

> SchemaView and in-built interactive docs will (likely) require this to be set back to `rest_framework.schemas.AutoSchema` to function correctly.

**Comment by AlanCoding** [comment]
Don't want to lose this

https://github.com/ansible/ansible/pull/62895

File: requirements/requirements.txt [hunk 1]
```diff
@@ -1,135 +1,132 @@
...
+django-polymorphic==2.1.2
 django-qsstats-magic==1.1.0
 django-radius==1.3.3
 django-solo==1.1.3
-django-split-settings==0.3.0
-django-taggit==0.22.2
-django==2.2.4
+django-split-settings==1.0.0
+django-taggit==1.2.0
+django==2.2.7
```
Comment (ryanpetrello) [inline requirements/requirements.txt]: 👀 https://github.com/advisories/GHSA-hvmf-r92r-27hr

(we should go with 2.2.8)

File: Makefile [hunk 2]
```diff
@@ -188,12 +190,13 @@ requirements_ansible_dev:
...
 	if [[ "$(PIP_OPTIONS)" == *"--no-index"* ]]; then \
-	    cat requirements/requirements.txt requirements/requirements_local.txt | $(VENV_BASE)/awx/bin/pip install $(PIP_OPTIONS) --ignore-installed -r /dev/stdin ; \
+	    cat requirements/requirements.txt requirements/requirements_local.txt | $(VENV_BASE)/awx/bin/pip install $(PIP_OPTIONS) --ignore-installed --no-build-isolation -r /dev/stdin ; \
 	else \
-	    cat requirements/requirements.txt requirements/requirements_git.txt | $(VENV_BASE)/awx/bin/pip install $(PIP_OPTIONS) --no-binary $(SRC_ONLY_PKGS) --ignore-installed -r /dev/stdin ; \
+	    cat requirements/requirements.txt requirements/requirements_git.txt | $(VENV_BASE)/awx/bin/pip install $(PIP_OPTIONS) --no-binary $(SRC_ONLY_PKGS) -r /dev/stdin ; \
 	fi
 	echo "include-system-site-packages = true" >> $(VENV_BASE)/awx/lib/python$(PYTHON_VERSION)/pyvenv.cfg
 	$(VENV_BASE)/awx/bin/pip uninstall --yes -r requirements/requirements_tower_uninstall.txt
+	rm $(shell ls -d $(VENV_BASE)/awx/lib/python* | head -n 1)/no-global-site-packages.txt
```
Comment (AlanCoding) [inline Makefile]: Still plan to document what this is doing with some comments

**Comment by AlanCoding** [comment]
May need to revisit the version of setuptools, or else mark as an upgrade blocker. We hit a complex versioning issue related to https://github.com/pypa/setuptools/issues/1916

**Comment by AlanCoding** [comment]
recheck

**Review by shanemcd — APPROVED** [review]

**Comment by AlanCoding** [comment]
recheck

**Pull request merged**